Emoji resource support.

CoreyShupe opened the pull request:
Addresses https://github.com/princesslana/ErisCasper.java/issues/125

CoreyShupe committed:
Emoji resource support.
Addresses https://github.com/princesslana/ErisCasper.java/issues/125

CoreyShupe committed:
Import fixing.

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/rest/EmojiRoute.java:
```diff
@@ -0,0 +1,71 @@
+   * @see <a href="https://discordapp.com/developers/docs/resources/emoji#list-guild-emojis">
+   *     https://discordapp.com/developers/docs/resources/emoji#list-guild-emojis</a>
+   */
+  public Route<Void, ImmutableList<Emoji>> getEmojis() {
```
I'd like to follow the names in the discord docs, so this should be listGuildEmojis.

By being consistent with the discord docs, we have to write less docs ourselves, because it's easier to find things in the discord docs.

CoreyShupe replied:
alright

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/rest/emoji/CreateEmojiRequest.java:
```diff
@@ -0,0 +1,20 @@
+public interface CreateEmojiRequest {
+  String getName();
+
+  EmojiImageWrapper getImage();
```
I'm undecided on this. I think my preferred way is to have this be String, and provide convenience methods that allow a File or whatever to be passed in.

CoreyShupe replied:
you can use a string in the current setup

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/util/FormatType.java:
```diff
@@ -0,0 +1,27 @@
+public enum FormatType {
+  GIF("gif"),
+  JPG("jpg"),
+  JPEG("jpeg"),
```
So, you said ImageIO supports both. What is the difference between them?

CoreyShupe replied:
not sure lemme look into it

ianagbip1oti commented on src/main/java/com/github/princesslana/eriscasper/util/SendableImage.java:
```diff
@@ -0,0 +1,31 @@
+  private SendableImage() {}
+
+  public static String imageToBase64(File file) throws IOException {
+    InputStream stream = file.toURI().toURL().openConnection().getInputStream();
```
This stream is never closed?

Look into Guava's IO stuff: https://github.com/google/guava/wiki/IOExplained. I think you'll want to start with `Files.asByteSource`

CoreyShupe committed:
Formatting changes + method remodel.
SendableImage changed how it reads bytes from a file.
CreateEmojiRequest now uses String and convenience methods for `getImage()`

CoreyShupe committed:
Discord docs consistency update.

CoreyShupe committed:
Removed non-needed image type `jpg`, we have `jpeg`.
